hs-probe-bsp/src/rcc.rs

```diff
@@ -111,22 +111,57 @@ impl RCC {
             PLLQ: pllq
         );
 
+        // Enable PWR domain and setup voltage scale and overdrive options
+        modify_reg!(rcc, self.rcc, APB1ENR, PWREN: Enabled);
+
+        let enable_overdrive;
+
+        // The scale can be modified only when the PLL is OFF and the
+        // HSI or HSE clock source is selected as system clock source.
+        let pwr = &*pwr::PWR;
+        if sysclk <= 144_000_000 {
+            modify_reg!(pwr, pwr, CR1, VOS: SCALE3);
+            enable_overdrive = false;
+        } else if sysclk <= 168_000_000 {
+            modify_reg!(pwr, pwr, CR1, VOS: SCALE2);
+            enable_overdrive = false;
+        } else if sysclk <= 180_000_000 {
+            modify_reg!(pwr, pwr, CR1, VOS: SCALE1);
+            enable_overdrive = false;
+        } else {
+            modify_reg!(pwr, pwr, CR1, VOS: SCALE1);
+            enable_overdrive = true;
+        }
+
         // Turn on PLL
         modify_reg!(rcc, self.rcc, CR, PLLON: On);
+
+        if enable_overdrive {
+            // Enable the over-drive mode
+            modify_reg!(pwr, pwr, CR1, ODEN: 1);
+            while read_reg!(pwr, pwr, CSR1, ODRDY) == 0 {}
+
+            // Switch the voltage regulator from normal mode to over-drive mode
+            modify_reg!(pwr, pwr, CR1, ODSWEN: 1);
+            while read_reg!(pwr, pwr, CSR1, ODSWRDY) == 0 {}
+        }
+
         // Wait for PLL to be ready
         while read_reg!(rcc, self.rcc, CR, PLLRDY == NotReady) {}
 
         // Adjust flash wait states
-        modify_reg!(flash, &*flash::FLASH, ACR, LATENCY: flash_latency);
+        let flash = &*flash::FLASH;
+        modify_reg!(flash, flash, ACR, LATENCY: flash_latency);
+
+        // Enable ART and prefetch
+        modify_reg!(flash, flash, ACR, ARTEN: Enabled);
+        modify_reg!(flash, flash, ACR, PRFTEN: Enabled);
```

Comment:
Since we access flash through AXIM (0x0800_0000 addresses) instead of ITCM (0x0020_0000 addresses), the ART and prefetch units cannot be used at all, so I don't think there's any point enabling it. If we did, though, you can do it in one modify_reg call instead...

Did you measure performance with AXIM vs ITCM+ART+prefetch?

![image](https://user-images.githubusercontent.com/47219/111071712-5ee38d80-84cf-11eb-89e4-999ed84a086d.png)


```suggestion
        modify_reg!(flash, flash, ACR, ARTEN: Enabled, PRFTEN: Enabled);
```

Reply:
I measured, overall probe performance was the same, but... I forgot that we should enable caches. This makes a real difference!

Reply:
Maybe we can remove ART+prefetch setup since we don't use them.

Reply:
We should be very careful enabling the dcache, all DMA operations will need cache management added. Do we still get the performance boost from just icache?

Reply:
Here is the data @Disasm sent me, the difference is huge!

```
no cache
Time elapsed: 2.924413305s, throughput is 27.356 Mbit/s

i-cache enabled
Time elapsed: 2.118092055s, throughput is 37.770 Mbit/s

i-cache & d-cache enabled
Time elapsed: 881.121465ms, throughput is 90.793 Mbit/s
```

Reply:
This is for a synthetic test. For hs-probe there is no significant difference between "i-cache" and "no-i-cache" options, but there is a difference between "i-cache" and "i-cache+d-cache" ones.

Reply:
D-cache indeed breaks everything, ecpdap even fails to scan JTAG chain. Reverted.

Reply:
Yea, that figures, I guess the JTAG DMA operation is breaking. I think some of the USB memory access would also be affected. I don't think it would be that hard to make it correct (but we might find some of the performance benefit goes away once it's done properly...) but perhaps it can be a second PR.

Reply:
USB itself works fine with D-cache because it doesn't use DMA and peripheral accesses shouldn't be cached.

Reply:
Ah, I see, the USB peripheral does have a DMA master but we don't use it, and since the USB FIFO memory is in the peripheral address space it's also marked non-cached for the dcache, so enabling it shouldn't have any effect there.

I guess that just leaves the DMA for the SWO, USART, and JTAG modes.